task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -229,9 +263,8 @@ impl SpHandler for MgsHandler {
         }));
 
         match update.component {
-            SpComponent::SP_ITSELF => self.common.update_prepare(update),
             SpComponent::HOST_CPU_BOOT_FLASH => {
-                self.host_flash_update.prepare(update)
+                self.host_flash_update.prepare(&UPDATE_MEMORY, update)
```

Comment:
It looks like `update_prepare` for the SP is moved to a separate MGS message type.  Do we need to remove `SP_ITSELF` from `enum SpComponent` [here](https://github.com/oxidecomputer/omicron/blob/7124e0a8381f26c767964664751d0edf7de32598/gateway-messages/src/lib.rs#L510-L511)?

Reply:
Ah, I guess not, because we still use it in `update_chunk`.  In that case, out of curiosity, why did `sp_update_prepare` diverge from `component_update_prepare`?

Reply:
`SpUpdatePrepare` grew fields that are auxflash specific (the auxflash image size and CHCK value), and does not have a component field (it's implicitly `SP_ITSELF`); `ComponentUpdatePrepare` has a `component` field but no auxflash fields.